src/PathFilter.cpp

```diff
@@ -0,0 +1,91 @@
+#include "../includes/PathFilter.h"
+
+#pragma unmanaged
+
+static const std::string doubleStarPrefix("**/");
+
+static bool prefix(const std::string &prefix, const std::string &str) {
+  return str.length() >= prefix.length() && std::equal(prefix.begin(), prefix.end(), str.begin());
+}
+
+static const std::string ensureTrailingSeparator(const std::string &path) {
+  if (path.back() == '/') {
+    return path;
+  } else {
+    return path + '/';
+  }
+}
+
+static const std::string normalizeTail(const std::string &path) {
+  int length = path.length();
+  if (length == 0) {
+    return path;
+  }
+  // Find the first '*' and remove trailing elements
+  auto istar = path.find('*');
+  if (istar != std::string::npos) {
+    return path.substr(0, istar);
+  }
+  // Path should end with exactly one '/' or none at all.
+  if (path[length - 1] == '/') {
+    if (length == 1) {
+      return path;
+    }
+    else {
+      int i = length - 2;
+      while (i >= 0 && path[--i] == '/');
+      return path.substr(0, i + 2);
+    }
+  }
+  return path + '/';
+}
+
+static void push_unique(std::vector<std::string> &vector, const std::string &element) {
+  for (auto e : vector) {
```

Comment:
```suggestion
static void pushUnique(std::vector<std::string> &vector, const std::string &element) {
  for (auto e : vector) {
```

Is there a standard of when to use `aB` versus `a_b` as a method name format?